--- a/meson.build
+++ b/meson.build
@@ -137,7 +137,7 @@
 libcall_ui_dep = libcall_ui.get_variable('libcall_ui_dep')
 libnm_dep = dependency('libnm', version: '>= 1.14')
 libpolkit_agent_dep = dependency('polkit-agent-1', version: '>= 0.105')
-libsystemd_dep = dependency('libsystemd', version: '>= 241')
+libsystemd_dep = dependency('libsystemd', version: '>= 241', required: false)
 network_agent_dep = dependency('libsecret-1')
 upower_glib_dep = dependency('upower-glib', version: '>=0.99.1')
 wayland_client_dep = dependency('wayland-client', version: '>=1.14')
@@ -157,6 +157,9 @@
 config_h.set_quoted('PHOSH_VERSION', meson.project_version())
 config_h.set('HAVE_RFKILL_EVENT_EXT', have_rfkill_event_ext)
 
+if libsystemd_dep.found()
+  config_h.set('HAVE_LIBSYSTEMD', 1)
+endif
 configure_file(
   input: 'config.h.in',
   output: 'config.h',
--- a/src/meson.build
+++ b/src/meson.build
@@ -280,7 +280,6 @@
   libhandy_dep,
   libnm_dep,
   libpolkit_agent_dep,
-  libsystemd_dep,
   network_agent_dep,
   upower_glib_dep,
   wayland_client_dep,
@@ -288,10 +287,17 @@
   cc.find_library('m', required: false),
   cc.find_library('rt', required: false),
 ]
-
+if libsystemd_dep.found()
+  phosh_deps += [libsystemd_dep]
+endif
+
+
+polkit_ver = libpolkit_agent_dep.version().split('.')
+polkit_major = polkit_ver[0]
+polkit_minor = polkit_ver.length() > 1 ? polkit_ver[1] : '0'
 add_project_arguments([
-  '-DPOLKIT_AGENT_MAJOR_VERSION=' + libpolkit_agent_dep.version().split('.')[0],
-  '-DPOLKIT_AGENT_MINOR_VERSION=' + libpolkit_agent_dep.version().split('.')[1]
+  '-DPOLKIT_AGENT_MAJOR_VERSION=' + polkit_major,
+  '-DPOLKIT_AGENT_MINOR_VERSION=' + polkit_minor,
 ], language: 'c')
 
 phosh_inc = include_directories('.')
--- a/src/main.c
+++ b/src/main.c
@@ -20,7 +20,9 @@
 #include <glib/gi18n.h>
 #include <glib-unix.h>
 
+#if HAVE_LIBSYSTEMD
 #include <systemd/sd-daemon.h>
+#endif
 
 static gboolean
 quit (gpointer unused)
@@ -79,7 +81,9 @@
   g_debug ("Phosh ready after %.2fs", g_timer_elapsed (timer, NULL));
 
   /* Inform systemd we're up */
+#if HAVE_LIBSYSTEMD
   sd_notify (0, "READY=1");
+#endif
 
   g_signal_handlers_disconnect_by_data (shell, timer);
 }
--- a/src/util.c
+++ b/src/util.c
@@ -12,7 +12,9 @@
 #include <locale.h>
 #include <glib/gi18n.h>
 
+#if HAVE_LIBSYSTEMD
 #include <systemd/sd-login.h>
+#endif
 
 #include <sys/mman.h>
 #include <sys/stat.h>
@@ -150,6 +152,7 @@
 phosh_find_systemd_session (char **session_id)
 {
   int n_sessions;
+#if HAVE_LIBSYSTEMD
 
   g_auto (GStrv) sessions = NULL;
   char *session;
@@ -196,6 +199,9 @@
     *session_id = g_strdup (session);
 
   return session != NULL;
+#else
+  return FALSE;
+#endif
 }
 
 /**
--- a/src/polkit-auth-agent.h
+++ b/src/polkit-auth-agent.h
@@ -18,8 +18,8 @@
 /* libpolkit lacks these */
 #if POLKIT_AGENT_MAJOR_VERSION == 0 && POLKIT_AGENT_MINOR_VERSION < 114
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitSubject, g_object_unref)
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitAgentListener, g_object_unref)
 #endif
-G_DEFINE_AUTOPTR_CLEANUP_FUNC(PolkitAgentListener, g_object_unref)
 
 #define PHOSH_TYPE_POLKIT_AUTH_AGENT (phosh_polkit_auth_agent_get_type())
 
